src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs

```diff
@@ -503,5 +518,44 @@ static bool CanConvertToSubpattern(IdentifierNameSyntax name, SemanticModel mode
                 }
             }
         }
+
+        protected override async Task FixAllAsync(
+            Document document,
+            ImmutableArray<TextSpan> fixAllSpans,
+            SyntaxEditor editor,
+            CodeActionOptionsProvider optionsProvider,
+            string? equivalenceKey,
+            CancellationToken cancellationToken)
+        {
+            // Get all the descendant nodes to refactor.
+            var nodes = editor.OriginalRoot.DescendantNodes().Where(IsFixableNode);
+
+            // We're going to be continually editing this tree. Track all the nodes we
+            // care about so we can find them across each edit.
+            document = document.WithSyntaxRoot(editor.OriginalRoot.TrackNodes(nodes));
+
+            // Process all nodes to refactor in reverse to ensure nested nodes
+            // are processed before the outer nodes to refactor.
+            foreach (var originalNode in nodes.Reverse())
+            {
+                // Only process nodes fully within a fixAllSpan
+                if (!fixAllSpans.Any(fixAllSpan => fixAllSpan.Contains(originalNode.Span)))
+                    continue;
+
+                // Get current root, current node to refactor and semantic model.
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var currentNode = root.GetCurrentNodes(originalNode).SingleOrDefault();
```

Comment:
you're using SingleOrDefault, but not checking the result?  should we check this for null?  or does GetReplacementFunc already do this?

Reply:
Yes, GetReplacementFunc handles null input.